File: sdk/rust/src/artifact/staticcheck.rs
use crate::{
    api::artifact::ArtifactSystem::{Aarch64Darwin, Aarch64Linux, X8664Darwin, X8664Linux},
    artifact::{language::go::Go, ArtifactSource},
    context::ConfigContext,
};
use anyhow::Result;

#[derive(Default)]
pub struct Staticcheck {}

impl Staticcheck {
    pub fn new() -> Self {
        Self::default()
    }

    pub async fn build(self, context: &mut ConfigContext) -> Result<String> {
        let name = "staticcheck";
        let source_version = "2025.1.1";
        let source_path = format!(
            "https://github.com/dominikh/go-tools/archive/refs/tags/{source_version}.tar.gz"
        );

        let source = ArtifactSource::new(name, source_path.as_str()).build();

        let build_directory = format!("go-tools-{source_version}");
        let build_path = format!("cmd/{name}/{name}.go");
        let systems = vec![Aarch64Darwin, Aarch64Linux, X8664Darwin, X8664Linux];

        Go::new(name, systems)
            .with_alias(format!("{name}:{source_version}"))
            .with_build_directory(build_directory.as_str())
            .with_build_path(build_path.as_str())
            .with_source(source)
            .build(context)
            .await
    }
}

File: sdk/typescript/src/artifact/staticcheck.ts
import { ArtifactSource } from "../artifact.js";
import type { ConfigContext } from "../context.js";
import { Go } from "./language/go.js";
import { SYSTEMS } from "../system.js";

/**
 * Builder for the staticcheck artifact.
 *
 * Mirrors Rust `Staticcheck` struct in `sdk/rust/src/artifact/staticcheck.rs`.
 * Builds staticcheck from the go-tools source using the Go language builder.
 */
export class Staticcheck {
  async build(context: ConfigContext): Promise<string> {
    const name = "staticcheck";
    const sourceVersion = "2026.1";
    const sourcePath = `https://sdk.vorpal.build/source/staticcheck-${sourceVersion}.tar.gz`;

    const source = new ArtifactSource(name, sourcePath).build();

    const buildDirectory = `go-tools-${sourceVersion}`;
    const buildPath = `cmd/${name}/${name}.go`;

    return new Go(name, SYSTEMS)
      .withAliases([`${name}:${sourceVersion}`])
      .withBuildDirectory(buildDirectory)
      .withBuildPath(buildPath)
      .withSource(source)
      .build(context);
  }
}
